File: backend/app/api/communication_groups.py
"""Compatibility exports for the migrated communication-groups HTTP adapter."""

from app.modules.identities.communication_groups.api import (
    create_communication_group,
    delete_communication_group,
    list_communication_groups,
    router,
    update_communication_group,
)

__all__ = [
    "create_communication_group",
    "delete_communication_group",
    "list_communication_groups",
    "router",
    "update_communication_group",
]

File: frontend/src/lib/api/communicationGroups.ts
import { apiFetch } from '@/lib/api/client';
import type {
  IdentityCommunicationGroupDTO,
  IdentityCommunicationGroupPayload,
} from '@/types';

export const listCommunicationGroups = () =>
  apiFetch<IdentityCommunicationGroupDTO[]>('/api/communication-groups');

export const createCommunicationGroup = (
  payload: IdentityCommunicationGroupPayload,
) =>
  apiFetch<IdentityCommunicationGroupDTO>('/api/communication-groups', {
    method: 'POST',
    body: JSON.stringify(payload),
  });

export const updateCommunicationGroup = (
  groupId: number,
  payload: IdentityCommunicationGroupPayload,
) =>
  apiFetch<IdentityCommunicationGroupDTO>(`/api/communication-groups/${groupId}`, {
    method: 'PUT',
    body: JSON.stringify(payload),
  });

export const deleteCommunicationGroup = (groupId: number) =>
  apiFetch<void>(`/api/communication-groups/${groupId}`, {
    method: 'DELETE',
  });
